kola/tests/crio/crio.go

```diff
@@ -0,0 +1,197 @@
+		Distros:     []string{"rhcos"},
+	})
+
+}
+
+// genContainer makes a container out of binaries on the host. This function uses podman
+func genContainer(c cluster.TestCluster, m platform.Machine, name string, binnames []string) {
+	cmd := `tmpdir=$(mktemp -d); cd $tmpdir; echo -e "FROM scratch\nCOPY . /" > Dockerfile;
+	        b=$(which %s); libs=$(sudo ldd $b | grep -o /lib'[^ ]*' | sort -u);
+	        sudo rsync -av --relative --copy-links $b $libs ./;
+	        sudo podman build -t %s .`
+
+	c.MustSSH(m, fmt.Sprintf(cmd, strings.Join(binnames, " "), name))
+}
+
+// crioBaseTests executes multiple tests under the "base" name
+func crioBaseTests(c cluster.TestCluster) {
+	c.Run("crio-info", testCrioInfo)
+	c.Run("networks-reliably", crioNetworksReliably)
+}
+
+// crioNetwork ensures that crio containers can make network connections outside of the host
+func crioNetwork(c cluster.TestCluster) {
+	machines := c.Machines()
+	src, dest := machines[0], machines[1]
+
+	c.Log("creating ncat containers")
+
+	genContainer(c, src, "ncat", []string{"ncat"})
+	genContainer(c, dest, "ncat", []string{"ncat"})
+
+	listener := func(ctx context.Context) error {
+		// Will block until a message is recieved
+		out, err := c.SSH(dest,
+			`echo "HELLO FROM SERVER" | podman run -i -p 9988:9988 ncat ncat --idle-timeout 20 --listen 0.0.0.0 9988`,
+		)
+		if err != nil {
+			return err
+		}
+
+		if !bytes.Equal(out, []byte("HELLO FROM CLIENT")) {
+			return fmt.Errorf("unexpected result from listener: %q", out)
+		}
+
+		return nil
+	}
+
+	talker := func(ctx context.Context) error {
+		// Wait until listener is ready before trying anything
+		for {
+			_, err := c.SSH(dest, "sudo lsof -i TCP:9988 -s TCP:LISTEN | grep 9988 -q")
+			if err == nil {
+				break // socket is ready
+			}
+
+			exit, ok := err.(*ssh.ExitError)
+			if !ok || exit.Waitmsg.ExitStatus() != 1 { // 1 is the expected exit of grep -q
+				return err
+			}
+
+			select {
+			case <-ctx.Done():
+				return fmt.Errorf("timeout waiting for server")
+			default:
+				time.Sleep(100 * time.Millisecond)
+			}
+		}
+
+		srcCmd := fmt.Sprintf(`echo "HELLO FROM CLIENT" | podman run -i ncat ncat %s 9988`, dest.PrivateIP())
+		out, err := c.SSH(src, srcCmd)
+		if err != nil {
+			return err
+		}
+
+		if !bytes.Equal(out, []byte("HELLO FROM SERVER")) {
+			return fmt.Errorf(`unexpected result from listener: "%v"`, out)
+		}
+
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	if err := worker.Parallel(ctx, listener, talker); err != nil {
+		c.Fatal(err)
+	}
+}
+
+// crioNetworksReliably verifies that crio containers have a reliable network
+func crioNetworksReliably(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	genContainer(c, m, "ping", []string{"sh", "ping"})
+
+	output := c.MustSSH(m, `for i in $(seq 1 100); do
+		echo -n "$i: "
+		podman run --rm ping sh -c 'ping -i 0.2 172.17.0.1 -w 1 >/dev/null && echo PASS || echo FAIL'
```

Comment:
I think this is going to require `sudo` since RHCOS/`kola` defaults to an unprivileged `core` user

Reply:
Good catch. @arithx is using `sudo` within these tests an OK pattern or is there something specific we should call when executing 'privileged' commands?

Reply:
I should have grepped :-). Looks like `sudo` is fine to use!

Reply:
Updated with `sudo`